diff --git a/gecko/gfx/layers/opengl/GrallocTextureHost.cpp b/gecko/gfx/layers/opengl/GrallocTextureHost.cpp
index 28cea35..3d34017 100644
--- a/gecko/gfx/layers/opengl/GrallocTextureHost.cpp
+++ b/gecko/gfx/layers/opengl/GrallocTextureHost.cpp
@@ -39,6 +39,7 @@ SurfaceFormatForAndroidPixelFormat(android::PixelFormat aFormat,
   case HAL_PIXEL_FORMAT_YCbCr_422_SP:
   case HAL_PIXEL_FORMAT_YCrCb_420_SP:
   case HAL_PIXEL_FORMAT_YCbCr_422_I:
+  case HAL_PIXEL_FORMAT_YCrCb_NV12:
   case GrallocImage::HAL_PIXEL_FORMAT_YCbCr_420_SP_TILED:
   case GrallocImage::HAL_PIXEL_FORMAT_YCbCr_420_SP_VENUS:
   case HAL_PIXEL_FORMAT_YV12:
